packages/nimble-components/src/table/components/cell/index.ts

```diff
@@ -14,10 +20,45 @@ declare global {
  * A styled cell that is used within the nimble-table-row.
  * @internal
  */
-export class TableCell extends FoundationElement {
+export class TableCell<
+    TCellData extends TableRecord = TableRecord
+> extends FoundationElement {
     // TODO: This should be replaced with an instance of TableCellState<TCellData>
     @observable
-    public data: TableFieldValue;
+    public data?: TableCellState<TCellData>;
+
+    @observable
+    public cellTemplate?: ViewTemplate;
+
+    @observable
+    public cellStyles?: ElementStyles;
+
+    private customCellView: HTMLView | undefined = undefined;
+
+    public override connectedCallback(): void {
+        super.connectedCallback();
+        this.updateView();
+    }
+
+    private updateView(): void {
+        const newCellView = this.customCellView === undefined;
+        if (newCellView) {
+            this.customCellView = this.cellTemplate!.create(this);
+            if (this.cellStyles) {
+                this.$fastController.addStyles(this.cellStyles);
+            }
+        }
+
+        this.customCellView?.bind(this.data, defaultExecutionContext);
+
+        if (newCellView) {
+            this.customCellView!.appendTo(this.shadowRoot!);
+        }
```

Comment:
I didn't try and dig into this but it seems odd for us to be directly manipulating the shadowRoot. This suggests that we can dynamically change the template using the `$fastController` for the element instance and I imagine that handles binding for you as well: https://www.fast.design/docs/fast-element/declaring-templates#templating-and-the-element-lifecycle

Reply:
I've added a `div` to the template to use as the host for the cell content, so we're not updating the `shadowRoot` of the custom element directly. I've also moved to leverage the FAST `render` method that simplifies the implementation a good deal. This is still needed so we can provide the proper binding source.